CMakeLists.txt

```diff
@@ -59,6 +59,8 @@ add_library(project_fp_options INTERFACE)
 
 add_library(project_warnings INTERFACE)
 
+add_library(turn_off_warnings INTERFACE)
```

Comment:
A new interface library, which does `-w` on gcc/clang and `/W0` on msvc to remove all warnings